cylc/flow/scheduler.py

```diff
@@ -1778,14 +1776,14 @@ def process_cylc_stop_point(self):
         """
         Set stop point.
 
-        In priority order stop cyclepoint (``stopcp``) is set:
-        * From the final point ff ``cylc restart --ignore-stop-cycle-point``.
-        * From the command line (``cylc <run/restart> --stop-point``).
+        In decreasing priority, stop cycle point (``stopcp``) is set:
+        * From the final point for ``cylc play --stopcp=ignore``.
+        * From the command line (``cylc play --stopcp=XYZ``).
         * From the database.
         * From the flow.cylc file (``[scheduler]stop after cycle point``).
```

Comment:
Actually, why should `cylc play --stopcp=ignore` (for a restart) use the final cycle point as the stop point, considering that `--icp=ignore` and `--fcp=ignore` result in use of the flow.cylc specified value?

Reply:
Sorry, I can't get my head around this! -"result in use of the flow.cylc specified value" of what exactly, the "stop CP"? 

The initial and final CPs define the extent of the graph, and start and stop CP specify where to start and stop within those bounds. So its seems reasonable to me to be able to ignore a previously-specified stop CP at restart, but shutdown at the FCP.  Or am I missing the point?

Reply:
Sorry for not being clear! If we're just talking about the pre-existing behaviour, which I don't think I've changed:

The behaviour of `--ignore-initial-cycle-point` is to ignore the icp stored in the database, and use whatever is specified by `[scheduling]initial cycle point`. Same for `--ignore-final-cycle-point`. But for `--ignore-stop-cycle-point`, while it ignores the stopcp in the database, it *doesn't* use `[scheduling]stop after cycle point`, it instead uses the fcp (not sure if from the database or flow.cylc).

So the behaviour is inconsistent between icp/fcp vs stopcp, is what I'm saying.

(For completeness, `--ignore-start-cycle-point` just ignores the startcp in the DB and doesn't do anything else. But as far as I can tell, the option won't make any difference to a restart though? The restart starts from whatever the last cycle point before shutdown was)

Reply:
Going to continue this discussion here: https://github.com/cylc/cylc-flow/issues/4062 (haven't written it up yet)

Reply:
(Sorry, missed this response a week ago ... but it's fine to punt it to the new issue)